codec.go

```diff
@@ -100,3 +101,27 @@ func FindEncoderByName(n string) *Codec {
 	defer C.free(unsafe.Pointer(cn))
 	return newCodecFromC(C.avcodec_find_encoder_by_name(cn))
 }
+
+func (c *Codec) HardwareConfigs(dt HardwareDeviceType) ([]CodecHardwareConfig, error) {
+	var configs []CodecHardwareConfig
+	var i int
+
+	for {
+		config := C.avcodec_get_hw_config(c.c, C.int(i))
+		if config == nil {
+			break
+		}
+
+		if HardwareDeviceType(config.device_type) == dt {
```

Comment:
Why are you checking this? 🤔

Reply:
we could skip this because we do it anyway in example when we range over